rumqttd/src/router/routing.rs

```diff
@@ -891,6 +973,51 @@ impl Router {
             }
         };
     }
+    fn send_all_alerts(&mut self) {
+        let span = tracing::info_span!("outgoing_alert");
+        let _guard = span.enter();
+
+        for (alert_id, (filter, alert_sender)) in &self.alerts {
+            trace!("Reading from alertlog: {}", filter);
+            let alerts = self
+                .alertlog
+                .native_readv(filter.to_string(), alert_id, 100)
+                .unwrap();
+            for alert in alerts {
+                alert_sender.send((alert_id, alert.0)).unwrap();
+            }
+        }
+    }
+}
+
+fn append_to_alertlog(alert: Alert, alertlog: &mut AlertLog) -> Result<Offset, RouterError> {
+    let topic = alert.topic();
+
+    let filter_idxs = alertlog.matches(&topic);
+
+    // Create a filter dynamically
+    let filter_idxs = match filter_idxs {
+        Some(v) => v,
+        None => {
+            let (idx, _cursor) = alertlog.next_native_offset(&topic);
```

Comment:
We should not be creating an AlertData for `topic` if it does not already exist in AlertLog. 

Although this will currently not happen since `alerts.topic()` always matches with some filter in AlertsLog, we should enforce otherwise also by omitting None branch altogether. 

Also, `filter_idxs` would never be `None` anyway since `alertlog.matches(..)` always returns a `Some(..)` so we can use .unwrap() on `filter_idxs`.

Reply:
fixed, also removed creation of unused filter